src/parser/parser.yy

```diff
@@ -2269,16 +2360,52 @@ show_config_item
     }
     ;
 
+// zone_list
+//     : zone {
+//         $$ = new ZoneList();
+//         $$->addZone($1);
+//     }
+//     | zone_list COMMA zone {
+//         $$ = $1;
+//         $$->addZone($3);
+//     }
+//     | zone_list COMMA {
+//         $$ = $1;
+//     }
+//     ;
+
+zone_name_list
+    : name_label {
+        $$ = new ZoneNameList();
+        $$->addZone($1);
+    }
+    | zone_name_list COMMA name_label {
+        $$ = $1;
+        $$->addZone($3);
+    }
+    ;
+
 create_space_sentence
     : KW_CREATE KW_SPACE opt_if_not_exists name_label {
         auto sentence = new CreateSpaceSentence($4, $3);
         $$ = sentence;
     }
+    // | KW_CREATE KW_SPACE opt_if_not_exists name_label KW_ON zone_list {
+    //     auto sentence = new CreateSpaceSentence($4, $3);
+    //     sentence->setZones($6);
+    //     $$ = sentence;
+    // }
     | KW_CREATE KW_SPACE opt_if_not_exists name_label L_PAREN space_opt_list R_PAREN {
         auto sentence = new CreateSpaceSentence($4, $3);
         sentence->setOpts($6);
         $$ = sentence;
     }
+    // | KW_CREATE KW_SPACE opt_if_not_exists name_label L_PAREN space_opt_list R_PAREN KW_ON zone_list {
+    //     auto sentence = new CreateSpaceSentence($4, $3);
+    //     sentence->setOpts($6);
+    //     sentence->setZones($9);
+    //     $$ = sentence;
+    // }
```

Comment:
Need some comment for the unimplemented features.

Reply:
yep  I will remove it and support later